cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -162,14 +295,31 @@ impl ClientTrait for MockClient {
         _system: System,
         _limit: u8,
     ) -> Result<SearchResults, SearchError> {
-        unimplemented!()
+        let mock_resp = self
+            .mock_responses
+            .lock()
+            .expect("couldn't acquire mock lock")
+            .pop_front();
+        if let Some(Response::Resolve(_)) = mock_resp {
+            panic!("found resolve response, expected search response");
+        } else if mock_resp.is_none() {
+            panic!("expected mock response, found nothing");
+        } else if let Some(Response::Error(err)) = mock_resp {
+            return Err(SearchError::Search(APIError::ErrorResponse(
+                err.try_into()?,
+            )));
+        } else if let Some(Response::Search(resp)) = mock_resp {
+            return Ok(resp);
+        }
+        return Err(MockDataError::InvalidData("unrecognized response".into()).into());
```

Comment:
suggestion: should'nt this panic as well?
I think including mock errors in the "production" error type is a bit leaky

Reply:
Fixed